### DISCUSSION TITLE
How to add name resolve to /etc/hosts for custom airflow docker image?

### DISCUSSION THREAD

**Discussion by woodlyer** [apache/airflow#34857] [Q&A]
When I custom my airflow image, I want to use name resolve with /etc/hosts. 
I try to add the text into /etc/hosts, but failed.

The docker image of airflow is strange.
```
FROM apache/airflow

USER root

RUN echo '192.168.1.10 hadoop01'  >> /etc/hosts && cat /etc/hosts
RUN echo "root:root" | chpasswd
```

Does anyone know how to do?
No root in the docker image, is very hard to do many things.

**Comment by wno-xyt** [comment]
You can do that dynamically while running you container using `--add-host` as described here: https://docs.docker.com/engine/reference/run/
or using compose as suggested here:
https://stackoverflow.com/questions/29076194/using-add-host-or-extra-hosts-with-docker-compose

**Reply by woodlyer** [reply]
How about doing this in helm chart?
